src/features/alerts/components/table-alerts.tsx

```diff
@@ -77,92 +74,79 @@ function IssueDetectedCellContent({ alert }: { alert: AlertConversation }) {
+    children: "Time",
+    width: 150,
+  },
+  {
+    id: "type",
+    children: "Type",
+    width: 150,
+  },
+  {
+    id: "event",
+    children: "Event",
+  },
+  {
+    id: "issue_detected",
+    children: "Issue detected",
+    width: 325,
+  },
+  {
+    id: "token_usage",
+    children: "Token usage",
+    width: 200,
+  },
+];
+
+function CellRenderer({
+  column,
+  row,
 }: {
-  hasMultipleWorkspaces: boolean;
+  column: Column;
+  row: AlertConversation;
 }) {
-  if (hasMultipleWorkspaces) {
-    return (
-      <div className="w-full flex flex-col items-center py-9 gap-2 px-4">
-        <IllustrationPackage className="size-36" />
-        <p className="font-bold text-4xl text-gray-900">No alerts found</p>
-        <p className="text-secondary text-xl">
-          Alerts will show up here when you use this workspace in your IDE
-        </p>
-        <LinkButton
-          href="https://docs.codegate.ai/features/workspaces"
-          target="_blank"
-          className="mt-4"
-        >
-          Learn about Workspaces
-          <LinkExternal02 />
-        </LinkButton>
-      </div>
-    );
-  }
+  switch (column.id) {
+    case "time":
+      return (
+        <span className="whitespace-nowrap">
+          {formatDistanceToNow(new Date(row.timestamp), {
+            addSuffix: true,
+          })}
+        </span>
+      );
+    case "type":
+      return <TypeCellContent alert={row} />;
+    case "event":
+      return getTitle(row);
+    case "issue_detected":
+      return (
+        <div className="truncate flex gap-2  items-center">
+          <IssueDetectedCellContent alert={row} />
+        </div>
+      );
+    case "token_usage":
+      return <TableAlertTokenUsage usage={row.conversation.token_usage_agg} />;
 
-  return (
-    <div className="w-full flex flex-col items-center py-9 gap-2 px-4">
-      <IllustrationDragAndDrop className="size-36" />
-      <p className="font-bold text-4xl text-gray-900">
-        Connect CodeGate to your IDE
-      </p>
-      <p className="text-secondary text-xl">
-        Learn how to get set up using{" "}
-        <Link
-          href="https://docs.codegate.ai/quickstart-continue"
-          target="_blank"
-          className="no-underline"
-        >
-          Continue
-        </Link>
-        ,{" "}
-        <Link
-          target="_blank"
-          href="https://docs.codegate.ai/quickstart"
-          className="no-underline"
-        >
-          Copilot
-        </Link>
-        , or{" "}
-        <Link
-          target="_blank"
-          href="https://docs.codegate.ai/how-to/use-with-aider"
-          className="no-underline"
-        >
-          Aider
-        </Link>
-        .
-      </p>
-      <LinkButton
-        href="https://docs.codegate.ai/"
-        target="_blank"
-        className="mt-4"
-      >
-        CodeGate Documentation
-        <LinkExternal02 />
-      </LinkButton>
-    </div>
-  );
+    default:
+      return column.id satisfies never;
```

Comment:
why not 


```suggestion
      throw new Error('tried to render non existent column')
```

Reply:
or use ts-pattern with `.exhaustive()` to get type safety guarantees that it does not happen